contrib/masterfiles-stage/masterfiles-stage.sh

```diff
@@ -1,45 +1,157 @@
+      exit 2
+      ;;
+    *)
+      # Assume that any other args should stay as-is
+      # (This would include short option flags as well
+      # as any other arguments passed to the script.)
+      params+=("$param")
+      ;;
+  esac
+done
+
+set -- "${params[@]}"
+
+# Set default variable values
+# (Don't change these without updating usage.)
+MASTERDIR=/var/cfengine/masterfiles
+PARAMS=/opt/cfengine/dc-scripts/params.sh
+verbose_mode=false
+debug_mode=false
+
+# Process using getopts
+while getopts ":d:p:vD" option; do
+  case "$option" in
+    d)
+      MASTERDIR="$OPTARG"
+      ;;
+    p)
+      PARAMS="$OPTARG"
+      ;;
+    v)
+      verbose_mode="true"
+      ;;
+    D)
+      debug_mode="true"
+      ;;
+    :)
+      echo "$0: Option -$OPTARG requires an argument"
+      usage
+      exit 3
+      ;;
+    *)
+      echo "$0: Invalid option: -$OPTARG"
+      usage
+      exit 2
+      ;;
+  esac
+done
+shift "$((OPTIND-1))"
+
+# Since we don't do anything with any remaining args, we should fail
+# noisily if we still have args left after the above shift command.
+[ "$#" -gt 0 ] && { echo "Too many arguments." ; usage ; exit 3 ; }
+
+if [ "$debug_mode" == "true" ]; then
+  echo "Option Deploy Dir: $MASTERDIR";
+  echo "Option Params File: $PARAMS";
 fi
 
-source $PARAMS
+# To test this option parsing file by itself, uncomment the below lines.
+# printf '%s\n' "MASTERDIR is $MASTERDIR"
+# printf '%s\n' "PARAMS is $PARAMS"
+# printf '%s\n' "verbose_mode is $verbose_mode"
+# printf '%s\n' "debug_mode is $debug_mode"
+################################################################################
+#                              END OPTION PARSING                              #
+################################################################################
 
-  # We probably want a different temporary location for each remote repository
-  # so that we can avoid conflicts and potential confusion.
-  # Example: 
-  # ROOT="/opt/cfengine/masterfiles_staging"
-  # PARAMS="/var/cfengine/policychannel/production_1.sh"
-  # TMP=/opt/cfengine/masterfiles/staging/_tmp_var_cfengine_policychannel_production_1_sh
+# Load common functionality, upstream implementations
+mydirname="$(dirname "$0")";
+source "${mydirname}/common.sh" || { echo "common.sh couldn't be sourced" >&2 ; exit 17 ; }
 
-  TRANSLATED_PARAMS=$(echo $PARAMS | tr [./] _)
-  STAGING_DIR="${ROOT}/_tmp${TRANSLATED_PARAMS}"
+[ -f "$PARAMS" ] || error_exit "ERROR: Missing '$PARAMS'"
+
+source "$PARAMS"
+
+  # The VCS_TYPE based function calls in the case switch below
+  # can count on the following environment variables to be set:
+  #
+  # MASTERDIR
+  # PARAMS
+  # VCS_TYPE (set in the PARAMS file)
+  # and additional vars set in PARAMS depending on the VCS_TYPE.
 
 case "${VCS_TYPE}" in
+    GIT_POLICY_CHANNELS)
+        git_stage_policy_channels
+        ;;
     GIT_TAG_OR_COMMIT)
-        git_tag_or_commit_masterstage
+        git_masterstage "${GIT_TAG_OR_COMMIT}"
```

Comment:
(GIT_TAG_OR_COMMIT as a vcs type could be deprecated entirely in favor of GIT, since the underlying code is now completely identical.)